### PR TITLE
Using WTF::String for blink data members

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
Upstream chromium has rolled out clang checks in upcoming versions that will error out in cases where blink headers are using `std::string` as data member type.

This change replaces all uses of `std::string` with `WTF::String` for data members in blink headers.

    Chromium change:
    https://chromium.googlesource.com/chromium/src/+/52472aba170b71f97ae9ab17f429ff494537adae
    
    commit 52472aba170b71f97ae9ab17f429ff494537adae
    Author: Xianzhu Wang <wangxianzhu@chromium.org>
    Date:   Mon Jan 9 20:26:58 2023 +0000
    
        Enable discouraged type check for data members under blink/renderer
    
        For existing usages of discouraged type:
        - Simple cases are changed to use blink types;
        - For other cases, ALLOW_DISCOURAGED_TYPE(reason) is added.
    
        Rules audit_non_blink_usages.py checking such usages are removed.
        We'll use the clang plugin from now on.
    
        Bug: 1363780
        Change-Id: I808524abc06c24d8f0f9867c06f4c6597bff1889
        Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4130490
        Commit-Queue: Xianzhu Wang <wangxianzhu@chromium.org>
        Reviewed-by: Daniel Cheng <dcheng@chromium.org>
        Reviewed-by: Chris Harrelson <chrishtr@chromium.org>
        Cr-Commit-Position: refs/heads/main@{#1090474}

Resolves https://github.com/brave/brave-browser/issues/28239

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: third_party/blink/renderer/core/brave_page_graph/page_graph.cc [hunk 1]
```diff
@@ -951,10 +952,10 @@ void PageGraph::GenerateReportForNode(const blink::DOMNodeId node_id,
 
   for (const GraphNode* succ : successors) {
     ItemName item_name = succ->GetItemName();
-    if (item_name.find("resource #") == 0) {
+    if (!item_name.Contains("resource #")) {
```
Comment (goodov) [inline third_party/blink/renderer/core/brave_page_graph/page_graph.cc:955]: `StartsWith`

**Review by goodov — APPROVED** [review]
lgtm with one fix

File: third_party/blink/renderer/core/brave_page_graph/page_graph.cc (hunk 1 above)
Comment (cdesouza-chromium) [reply third_party/blink/renderer/core/brave_page_graph/page_graph.cc]: Done

File: third_party/blink/renderer/core/brave_page_graph/page_graph.cc [hunk 2]
```diff
@@ -951,10 +952,10 @@ void PageGraph::GenerateReportForNode(const blink::DOMNodeId node_id,
 
   for (const GraphNode* succ : successors) {
     ItemName item_name = succ->GetItemName();
-    if (item_name.find("resource #") == 0) {
+    if (!item_name.StartsWith("resource #")) {
```
Comment (goodov) [inline third_party/blink/renderer/core/brave_page_graph/page_graph.cc:955]: why negation? it should be `if (item_name.StartsWith("resource #")) {`.

Comment (cdesouza-chromium) [reply third_party/blink/renderer/core/brave_page_graph/page_graph.cc]: Done. I had assumed I was replacing `count` there. Thanks for noticing that.

**Review by goodov — APPROVED** [review]

File: third_party/blink/renderer/core/brave_page_graph/graph_item/edge/event_listener/edge_event_listener_action.cc [hunk 3]
```diff
@@ -35,10 +36,12 @@ ScriptId EdgeEventListenerAction::GetListenerScriptId() const {
...
 
 ItemDesc EdgeEventListenerAction::GetItemDesc() const {
-  return GraphEdge::GetItemDesc() + " [" + event_type_ + "]" +
-         " [listener id: " + base::NumberToString(listener_id_) + "]" +
-         " [listener script id: " +
-         base::NumberToString(GetListenerScriptId()) + "]";
+  WTF::TextStream ts;
+  ts << GraphEdge::GetItemDesc() << " [" << event_type_ << "]"
+     << " [listener id: " << base::NumberToString(listener_id_) << "]"
+     << " [listener script id: " << base::NumberToString(GetListenerScriptId())
```
Comment (goodov) [inline third_party/blink/renderer/core/brave_page_graph/graph_item/edge/event_listener/edge_event_listener_action.cc:42]: one more thing: `base::NumberToString` should be removed as they should be natively supported by `WTF::TextStream`.

Comment (cdesouza-chromium) [reply third_party/blink/renderer/core/brave_page_graph/graph_item/edge/event_listener/edge_event_listener_action.cc]: Done. For this and all the other occurrences.

Comment (goodov) [reply third_party/blink/renderer/core/brave_page_graph/graph_item/edge/event_listener/edge_event_listener_action.cc]: what about `base/strings/string_number_conversions.h` includes?

Comment (cdesouza-chromium) [reply third_party/blink/renderer/core/brave_page_graph/graph_item/edge/event_listener/edge_event_listener_action.cc]: Done

**Pull request merged**